src/Resources/config/validators.xml

```diff
@@ -7,7 +7,7 @@
     <services>
         <service id="cmf_routing.validator.route_defaults" class="Symfony\Cmf\Bundle\RoutingBundle\Validator\Constraints\RouteDefaultsValidator">
             <argument id="controller_resolver" type="service"/>
-            <argument id="templating" type="service"/>
+            <argument id="twig.loader" type="service"/>
```

Comment:
i guess the real BC break is this. what if people use a non-twig template? then things will suddenly stop working. could we find a way to use templating if it exists, but twig.loader otherwise? maybe with an expression language check?

Reply:
I would suggest to do it like here:
https://github.com/symfony/framework-bundle/blob/master/Controller/ControllerTrait.php#L211
in a kind of compiler pass. We would have two validator classes then. If the user does a `composer require symfony/templating` a validator using templating would overwrite the default validator using twig in our container.

Reply:
But we have to document that `symfony/templating` should be added manually from Symfony 3.2 on, cause it was removed from frameworkbundle.

Reply:
+1 for the compiler pass to update the service definition for the validator to use the templating service if available. i would just remove the typehint and say it must be one or the other class.

we never required the templating here, so explaining how to update is "out of scope" and part of upgrading symfony itself. we can just explain what happened in the changelog, like " RoutingBundle can now also directly work with Twig. Symfony FrameworkBundle no longer requires symfony/templating since 3.2 and the component is deprecated. If the templating component is available in your application, it is however still used for BC."

Reply:
I would avoid to remove the type hint, cause the fact that both services got the same method is random.  So yes, we should have two services, which can be changed by a compiler pass.

to the templating problem as a dependency:
When having a Symfony App with version >= 3.2 and doing a simple `composer require symfony-cmf/routing-bundle` to add routing functions, you will get an error that templating service isn't available. The reason is: templating was removed from framework-bundle. So have to do a `composer require symfony/templating` before for those apps. So requiring templating is not a question of which templating engine to chose, it is a question of the symfony version.